apps/console/src/features/applications/api/use-scopes-of-api-resources.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { HttpMethods } from "@wso2is/core/models";
+import useRequest, {
+    RequestErrorInterface,
+    RequestResultInterface
+} from "apps/console/src/features/core/hooks/use-request";
+import { AxiosRequestConfig } from "axios";
+import { store } from "../../core/store";
+import { AuthorizedPermissionListItemInterface } from "../models/api-authorization";
+
+/**
+ * Hook to get the scopes of an API resource.
+ *
+ * @param appId - application ID.
+ * @returns SWR response object containing the data, error, isLoading, mutate.
+ * @throws `IdentityAppsApiException`
+ */
+const useScopesOfAPIResources = <Data = AuthorizedPermissionListItemInterface[], Error = RequestErrorInterface>(
```

Comment:
shall we move this to the `api-authoirzation.ts` file

Reply:
Moved the GET hooks to dedicated modules with the suggestion given in a previous PR (https://github.com/wso2/identity-apps/pull/4271/files#r1367843706).